File: 06-subprogramas/codes/valor.c
#include <stdio.h>

void incrementar(int numero) {
    printf("Dentro da funcao: O valor inicial do numero eh %d\n", numero);
    numero = numero + 10; // Altera a COPIA do valor
    printf("Dentro da funcao: O valor apos o incremento eh %d\n", numero);
}

int main() {
    int num = 5;
    printf("Antes da chamada da funcao: O valor do numero eh %d\n", num);

    incrementar(num); // Passa apenas uma COPIA de num

    printf("Depois da chamada da funcao: O valor do numero eh %d\n", num);
    return 0;
}

File: 06-subprogramas/codes/valor.java
public class valor {

    public static void incrementar(int numero) {
        System.out.println("Dentro do metodo (por valor): O valor inicial do numero eh " + numero);
        numero = numero + 10; // Altera a COPIA do valor
        System.out.println("Dentro do metodo (por valor): O valor apos a alteracao eh " + numero);
    }

    public static void main(String[] args) {
        int num = 5;
        System.out.println("Antes da chamada do metodo: O valor de meuNumero eh " + num);

        incrementar(num); // Passa uma COPIA de num

        System.out.println("Depois da chamada do metodo: O valor de meuNumero eh " + num); // num continua 5
    }
}